pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -368,6 +368,9 @@ func (a *Assignment) findFlavorForResourceGroup(
 			}
 		}
 
+		if cq.Preemption.PreemptBasedOnFlavorOrder && representativeMode == Preempt {
```

Comment:
The idea should be that if preemption is not possible, we proceed to the next flavor. But the flavor assigner doesn't really know about whether preemption is possible.
We might need to change the implementation of framework assigner to return a list of assignments that the scheduler and preemption algorithm. Or maybe we need to add tighter dependencies between the flavorAssigner and the preemption algorithms.

Reply:
I think we can just keep track in-memory of whether preemption was possible. If it was not, we put the workload back into the head of the queue to try the next flavor in the next iteration. And once we have passed through all the flavors, we requeue the workload as inadmissible.

Reply:
Thanks for your time and help. I will continue to improve this submission. 😸

Reply:
consider writing a design for this feature, following this template https://github.com/kubernetes-sigs/kueue/tree/main/keps/NNNN-template

Reply:
No problem, I will submit a proposal in next week